ledger/participant-integration-api/src/main/scala/platform/store/backend/StorageBackend.scala

```diff
@@ -13,13 +13,14 @@ trait StorageBackend[DB_BATCH] {
   def batch(dbDtos: Vector[DBDTOV1]): DB_BATCH
   def insertBatch(connection: Connection, batch: DB_BATCH): Unit
   def updateParams(connection: Connection, params: StorageBackend.Params): Unit
-  def initialize(connection: Connection): StorageBackend.Initialized
+  def initialize(connection: Connection): StorageBackend.LedgerEnd
+  def ledgerEnd(connection: Connection): StorageBackend.LedgerEnd
 }
 
 object StorageBackend {
   case class Params(ledgerEnd: Offset, eventSeqId: Long, configuration: Option[Array[Byte]])
 
-  case class Initialized(lastOffset: Option[Offset], lastEventSeqId: Option[Long])
+  case class LedgerEnd(lastOffset: Option[Offset], lastEventSeqId: Option[Long])
```

Comment:
```suggestion
  final case class LedgerEnd(lastOffset: Option[Offset], lastEventSeqId: Option[Long])
```
Same for Params above

Reply:
Maybe I missing here something? Why put final here and not everywhere else?
Scala is broad and you could do OO in scala yes, but where I come from I wouldn't ever do extend from a class, there are so many nicer functional ways to compose. It was some time ago I was working with hardcore OO language like java, but if I remember correctly even there is the saying to prefer composition to inheritance. So I sort of assumed in scala it is also ok to follow the same principle (even if not abandoning the whole OO like entirely ;) ), which would make all these final-s an unnecessary clutter to me.
Especially in case classes inheritance can be tricky with the hashcode equals contract, so AFAIK it is an antipattern.